### PR TITLE
Refactor!: move project from snapshot to model/node

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlmesh/core/model/decorator.py [hunk 1]
```diff
@@ -58,6 +58,7 @@ def model(
         dialect: t.Optional[str] = None,
         time_column_format: str = c.DEFAULT_TIME_COLUMN_FORMAT,
         physical_schema_override: t.Optional[t.Dict[str, str]] = None,
+        project: str = "",
```
Comment (izeigerman) [inline sqlmesh/core/model/decorator.py:61]: Should this be Optional rather than empty string?

File: sqlmesh/core/node.py [hunk 2]
```diff
@@ -154,6 +155,7 @@ class Node(PydanticModel):
     """
 
     name: str
+    project: str = ""
```
Comment (izeigerman) [inline sqlmesh/core/node.py:158]: Ditto: should this be optional?

File: sqlmesh/core/model/decorator.py (hunk 1 above)
Comment (tobymao) [reply sqlmesh/core/model/decorator.py:61]: i wanted to always be filled do you have a strong pref?

**Review by izeigerman — APPROVED** [review]
One question, LGTM otherwise

File: sqlmesh/core/model/decorator.py (hunk 1 above)
Comment (izeigerman) [reply sqlmesh/core/model/decorator.py:61]: Well, empty string is not exactly meaningful, so None is at least explicit. Up to you

**Pull request merged**